src/Appwrite/Platform/Workers/Deletes.php

```diff
@@ -516,13 +516,8 @@ private function deleteUser(callable $getProjectDB, Document $document, Document
             if ($document->getAttribute('confirm')) { // Count only confirmed members
                 $teamId = $document->getAttribute('teamId');
                 $team = $dbForProject->getDocument('teams', $teamId);
-                if (!$team->isEmpty()) {
-                    $team = $dbForProject->updateDocument(
-                        'teams',
-                        $teamId,
-                        // Ensure that total >= 0
-                        $team->setAttribute('total', \max($team->getAttribute('total', 0) - 1, 0))
-                    );
+                if (!$team->isEmpty() && $team->getAttribute('total', 0) > 0) {
+                    $dbForProject->decreaseDocumentAttribute('teams', $teamId, 'total', 1);
```

Comment:
to be picky, this is not atomic 😁 however I think we are missing conditional operations for this on the db library.